python/cudf/cudf/core/buffer/buffer.py

```diff
@@ -275,19 +255,26 @@ def get_ptr(self, *, mode) -> int:
             Failure to fulfill this contract will cause
             incorrect behavior.
 
+        Returns
+        -------
+        int
+            The device pointer as an integer
 
         See Also
         --------
         SpillableBuffer.get_ptr
-        CopyOnWriteBuffer.get_ptr
+        TenableBuffer.get_ptr
         """
         return self._ptr
 
-    def memoryview(self) -> memoryview:
+    def memoryview(
+        self, *, offset: int = 0, size: Optional[int] = None
+    ) -> memoryview:
         """Read-only access to the buffer through host memory."""
-        host_buf = host_memory_allocation(self.size)
+        size = self._size if size is None else size
+        host_buf = host_memory_allocation(size)
         rmm._lib.device_buffer.copy_ptr_to_host(
-            self.get_ptr(mode="read"), host_buf
+            self.get_ptr(mode="read") + offset, host_buf
         )
```

Comment:
Does anyone do error checking if we ask for a larger memoryview than is allocated?

Reply:
no, I don't think so